universal-application-tool-0.0.1/app/repository/AmazonS3Client.java

```diff
@@ -0,0 +1,101 @@
+package repository;
+
+import com.typesafe.config.Config;
+import java.util.concurrent.CompletableFuture;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.inject.ApplicationLifecycle;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+@Singleton
+public class AmazonS3Client {
+  public static final String AWS_S3_BUCKET = "aws.s3.bucket";
+  private static final Logger log = LoggerFactory.getLogger("s3client");
+
+  private final ApplicationLifecycle appLifecycle;
+  private final Config config;
+  private String bucket;
+  private S3Client s3;
+
+  @Inject
+  public AmazonS3Client(ApplicationLifecycle appLifecycle, Config config) {
+    this.appLifecycle = appLifecycle;
+    this.config = config;
+
+    log.info("aws s3 enabled: " + String.valueOf(enabled()));
+    if (enabled()) {
+      connect();
+      putTestObject();
+      getTestObject();
+    }
+
+    appLifecycle.addStopHook(
+        () -> {
+          if (s3 != null) {
+            s3.close();
+          }
+          return CompletableFuture.completedFuture(null);
+        });
+  }
+
+  public boolean enabled() {
+    return config.hasPath(AWS_S3_BUCKET);
+  }
+
+  public void putObject(String key, byte[] data) {
+    throwIfUninitialized();
```

Comment:
*suggestion*: I mildly prefer a conditional reinitialize, which is a little more forgiving.

Reply:
Sure thing!